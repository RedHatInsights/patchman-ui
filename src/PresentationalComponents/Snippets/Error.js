import { Card, EmptyState, EmptyStateBody, EmptyStateIcon, EmptyStateVariant, Title } from '@patternfly/react-core';
import FrownOpenIcon from '@patternfly/react-icons/dist/js/icons/frown-open-icon';
import propTypes from 'prop-types';
import React from 'react';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';

const Error = ({ message }) =>
    <Card>
        <EmptyState variant={EmptyStateVariant.full}>
            <EmptyStateIcon icon={FrownOpenIcon} color="black"/>
            <Title headingLevel="h5" size="lg">
                {intl.formatMessage(messages.statesError)}
            </Title>
            <EmptyStateBody>
                {message}
            </EmptyStateBody>
        </EmptyState></Card>;

export default Error;

Error.propTypes = {
    message: propTypes.string
};
