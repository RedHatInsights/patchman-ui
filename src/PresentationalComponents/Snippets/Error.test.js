import Error from './Error';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Error.js', () => {
    it('Should match the snapshot', () => {
        const wrapper = shallow(<Error message = {'testMessage'} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

