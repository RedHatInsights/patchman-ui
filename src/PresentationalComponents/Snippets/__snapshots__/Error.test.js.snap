// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error.js Should match the snapshot 1`] = `
<Card>
  <EmptyState
    variant="full"
  >
    <EmptyStateIcon
      color="black"
      icon={[Function]}
    />
    <Title
      headingLevel="h5"
      size="lg"
    >
      Error
    </Title>
    <EmptyStateBody>
      testMessage
    </EmptyStateBody>
  </EmptyState>
</Card>
`;
